### ISSUE TITLE
Camunda Modeler 5.16 Task Inputparmeter Type List alphabetic sorted

### ISSUE THREAD

**Issue by detmar-krause** [camunda/camunda-modeler#3978]
### Describe the bug

When creating a service task and adding a Inputs parameter as List

![nonSorted jpg](https://github.com/camunda/camunda-modeler/assets/57713315/578b4462-b85b-40d1-bad1-66a812f859f2)

and switch to an other task and switch back and open parameter the list ist sorted alphabatically

![sorted](https://github.com/camunda/camunda-modeler/assets/57713315/0cab9adc-57ea-44ee-b6c8-3b023263e367)

also the bpmn xml
```
             <camunda:inputParameter name="exec_parameters">
                <camunda:list>
                  <camunda:value>SARImageTransformer -joborder ${JSON(imageTransformerContainerJoborderBindFolder).prop("joborderBindFolder").value()}/${imageTransformerJoborderFileName}</camunda:value>
                  <camunda:value>-c</camunda:value>
                  <camunda:value>bash</camunda:value>
                </camunda:list>
              </camunda:inputParameter>
```

### Steps to reproduce

1. add parameter to Inputs as List
2. add List elements 
3. leave task
4. open properties panel of task again
5. open parameter and expand List elements
6. Elements are sorted

### Expected behavior

Parameter elements of a List should never be sorted or changed in order. 

### Environment

- OS: Ubuntu
- Camunda Modeler Version: 5.16 + 5.17
- Execution Platform:Camunda 7

----

Depends on https://github.com/bpmn-io/bpmn-js-properties-panel/pull/1047

**Comment by nikku** [comment]
Thanks for opening this bug report.

**Comment by nikku** [comment]
This may be addressed at some point via this (stalled) PR: https://github.com/bpmn-io/bpmn-js-properties-panel/pull/875.

**Comment by smbea** [comment]
I'm assigning to you @nikku since you already had a look. Feel free to un-assign if you can't look into it

**Comment by nikku** [comment]
Moving this issue back to `backlog` as we did not find the time to wrap it up.

**Comment by AlexanderSkrock** [comment]
Hi there,
I would like to ask about the current state of this issue. Also I had a look into the referenced draft pull request, but I was not able to find out what's missing.

@nikku , if you could potentially spare some time to wrap it up for me, I would volunteer to further work on this issue.

Kind regards
Alexander Skrock

**Comment by nikku** [comment]
@AlexanderSkrock we decided to remove alphabetic sorting all together. It does more harm than good and we regret that past decision today :see_no_evil:.

There exists a draft PR (https://github.com/bpmn-io/bpmn-js-properties-panel/issues/875) which is stuck, because the topic caused a number of nasty side-effects on other features of the properties panel (such as focussing entries). 

If you want to take a look into that and assess if it is something you'd want to contribute, please feel free to do so.

**Comment by AlexanderSkrock** [comment]
So, just to verify that I get you right, @nikku , should we set `shouldSort` to `false` by default as mention in your linked issue and then analyze occuring focus issues or did you mean to completely remove the option to use sorting?

Either way, I would give it a try, when I have some spare time.

**Comment by nikku** [comment]
We want to remove `shouldSort` completely:

* If a consumer uses it print a `console.warn` that it is no longer supported
* Automatically _all_ items stop to sort, in all places in the properties panel
* If sorting is desired, then on the domain level you _have to_ sort your inputs (list of variables, ...)

**Comment by nikku** [comment]
This has been implemented via https://github.com/bpmn-io/bpmn-js-properties-panel/pull/1047 and needs to be integrated into the modeler.

To be shipped with v5.24.0 :slightly_smiling_face:
